pkg/master/controller/crdregistration/crdregistration_controller.go

```diff
@@ -208,19 +222,48 @@ func (c *crdRegistrationController) handleVersionUpdate(groupVersion schema.Grou
 				continue
 			}
 
-			c.apiServiceRegistration.AddAPIServiceToSync(&apiregistration.APIService{
+			apiService := &apiregistration.APIService{
 				ObjectMeta: metav1.ObjectMeta{Name: apiServiceName},
 				Spec: apiregistration.APIServiceSpec{
 					Group:                groupVersion.Group,
 					Version:              groupVersion.Version,
 					GroupPriorityMinimum: 1000, // CRDs should have relatively low priority
 					VersionPriority:      100,  // CRDs will be sorted by kube-like versions like any other APIService with the same VersionPriority
 				},
-			})
+			}
+			c.apiServiceRegistration.AddAPIServiceToSync(apiService)
+			// Serve CRD OpenAPI spec if we have non-nil OpenAPIV3Schema
+			if c.openAPIAggregationManager != nil &&
```

Comment:
Yes. This serve-openapi-spec part lives in `crdRegistrationController.handleVersionUpdate()`, which gets called by `crdRegistrationController.processNextWorkItem()`. The work queue consists of all CRD adding/updating/deleting events. 

When a validation schema in a CRD changes, this part re-constructs the entire swagger (spec) for the CRD and calls `c.openAPIAggregationManager.AddUpdateLocalAPIServiceSpec()`. (Arguably it can instead read the existing spec and just update the schema, which sounds more efficient. But the construction part is just filling up a template, so I don't expect it to introduce much overhead)